src/plugins/cloudMetadata.nim

```diff
@@ -23,40 +23,178 @@ const
+      isSubscribedKey("_OP_CLOUD_PROVIDER_INSTANCE_TYPE"):
+    trace("Querying for GCP metadata")
+    if isSubscribedKey("_GCP_PROJECT_METADATA"):
+      let projectOpt = hitProviderEndpoint("http://169.254.169.254/computeMetadata/v1/project/?recursive=true", newHttpHeaders([("Metadata-Flavor", "Google")]))
+      if projectOpt.isSome():
+        try:
+          let valueProj = projectOpt.get()
+          if valueProj.startswith("{"):
+            let jsonProjValue = parseJson(valueProj)
+            setIfNeeded(result, "_GCP_PROJECT_METADATA", jsonProjValue.nimJsonToBox())
+          else:
+            trace("GCP project metadata didnt respond with json object. Ignoring it")
+        except:
+          trace("Could not insert _GCP_PROJECT_METADATA")
+
+    let resultOpt = hitProviderEndpoint("http://169.254.169.254/computeMetadata/v1/instance/?recursive=true", newHttpHeaders([("Metadata-Flavor", "Google")]))
+    if not resultOpt.isSome():
+      trace("Did not get instance metadata back from GCP endpoint")
+      return
+    let value = resultOpt.get()
+    if not value.startswith("{"):
+      trace("GCP metadata didnt respond with json object. Ignoring it")
+      return
+    try:
+      let jsonValue = parseJson(value)
+      try:
+        setIfNeeded(result, "_GCP_INSTANCE_METADATA", jsonValue.nimJsonToBox())
+      except:
+        trace("Could not insert _GCP_INSTANCE_METADATA")
+      try:
+        setIfNeeded(result, "_OP_CLOUD_PROVIDER_TAGS", jsonValue["tags"].nimJsonToBox())
+      except:
+        trace("Could not insert _OP_CLOUD_PROVIDER_TAGS for gcp")
+      try:
+        setIfNeeded(result, "_OP_CLOUD_PROVIDER_ACCOUNT_INFO", jsonValue["serviceAccounts"].nimJsonToBox())
+      except:
+        trace("Could not insert _OP_CLOUD_PROVIDER_TAGS for gcp")
+      try:
+        for iface in jsonValue["networkInterfaces"]:
+          var found = false
+          for config in iface["accessConfigs"]:
+            let ipv4 = config["externalIp"].getStr()
+            if ipv4 != "":
+              found = true
+              # just pick the first
+              setIfNeeded(result, "_OP_CLOUD_PROVIDER_IP", ipv4)
+              break
+          if found:
+            break
+      except:
+        trace("Could not insert _OP_CLOUD_PROVIDER_IP for gcp")
+      try:
+        setIfNeeded(result, "_OP_CLOUD_PROVIDER_REGION", jsonValue["zone"].getStr().split("/")[^1])
+      except:
+        trace("Could not insert _OP_CLOUD_PROVIDER_REGION for gcp")
+      try:
+        setIfNeeded(result, "_OP_CLOUD_PROVIDER_INSTANCE_TYPE", jsonValue["machineType"].getStr().split("/")[^1])
+      except:
+        trace("Could not insert _OP_CLOUD_PROVIDER_INSTANCE_TYPE for gcp")
+    except:
+      trace("GCP metadata responded with invalid json")
+
+    if getEnv(K_SERVICE) != "" and getEnv(CLOUD_RUN_TIMEOUT_SECONDS) != "":
+      result["_OP_CLOUD_PROVIDER_SERVICE_TYPE"] = pack("gcp_cloud_run_service")
+
+proc getAwsToken(): Option[string] =
```

Comment:
noticed you moved this down but can we keep all the http interaction functions next to each other?